File: src/test/java/io/bloco/faker/components/PhoneNumberTest.java
package io.bloco.faker.components;

import org.junit.Before;
import org.junit.Test;

import io.bloco.faker.Faker;

import static org.hamcrest.core.Is.is;
import static org.hamcrest.core.IsEqual.equalTo;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertThat;

public class PhoneNumberTest {

    private Faker faker;

    @Before
    public void setUp() {
        faker = new Faker("en-US");
    }

    @Test
    public void phoneNumber() {
        assertNotNull(faker.phoneNumber.phoneNumber());
    }

    @Test
    public void cellPhone() {
        assertNotNull(faker.phoneNumber.cellPhone());
    }

    @Test
    public void areaCode() {
        assertNotNull(faker.phoneNumber.areaCode());
    }

    @Test
    public void exchangeCode() {
        assertNotNull(faker.phoneNumber.exchangeCode());
    }

    @Test
    public void subscriberNumberDefault() {
        assertNotNull(faker.phoneNumber.subscriberNumber());
    }

    @Test
    public void subscriberNumber() {
        assertThat(faker.phoneNumber.subscriberNumber(10).length(), is(equalTo(10)));
    }

    @Test
    public void extensionDefault() {
        assertNotNull(faker.phoneNumber.subscriberNumber());
    }

    @Test
    public void extension() {
        assertThat(faker.phoneNumber.extension(10).length(), is(equalTo(10)));
    }
}

File: lib/src/test/kotlin/io/bloco/faker/components/PhoneNumberTest.kt
package io.bloco.faker.components

import io.bloco.faker.Faker
import junit.framework.TestCase.assertNotNull
import junit.framework.TestCase.assertTrue
import org.junit.Test

class PhoneNumberTest {
    private val faker = Faker("en-US")

    @Test
    fun phoneNumber() {
        assertNotNull(faker.phoneNumber.phoneNumber())
    }

    @Test
    fun cellPhone() {
        assertNotNull(faker.phoneNumber.cellPhone())
    }

    @Test
    fun areaCode() {
        assertNotNull(faker.phoneNumber.areaCode())
    }

    @Test
    fun exchangeCode() {
        assertNotNull(faker.phoneNumber.exchangeCode())
    }

    @Test
    fun subscriberNumberDefault() {
        assertNotNull(faker.phoneNumber.subscriberNumber())
    }

    @Test
    fun subscriberNumber() {
        assertTrue(faker.phoneNumber.subscriberNumber(10).length == 10)
    }

    @Test
    fun extensionDefault() {
        assertNotNull(faker.phoneNumber.subscriberNumber())
    }

    @Test
    fun extension() {
        assertTrue(faker.phoneNumber.extension(10).length == 10)
    }
}
